xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -394,6 +394,25 @@ int CSelectionStreams::IndexOf(StreamType type, int source, int64_t demuxerId, i
     return -1;
 }
 
+int CSelectionStreams::IndexOf(StreamType type, int type_id) const
```

Comment:
please drop this.

Reply:
Then I either have to adapt the other IndexOf, allowing demuxId == -1 in case of a nav source, or add a demuxId in Update, like I did in the other pr. Which one do you prefer?

Reply:
I think setting demuxerId in update is best option.